### PR TITLE
[Compute] Fix issue #10728 az vm create - option --subnet having inco…

### PR DESCRIPTION
**Pull request by arrownj**
…rrect behavior

Fix issue #10728

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -613,9 +613,30 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
...
+                client = get_network_client(cmd.cli_ctx).virtual_networks
+                vnet = cached_get(cmd, client.get, resource_group_name, vnet_name)
+
+                Subnet = cmd.get_models('Subnet', resource_type=ResourceType.MGMT_NETWORK)
+                subnet_obj = Subnet(
+                    name=subnet,
+                    address_prefixes=[subnet_address_prefix],
+                    address_prefix=subnet_address_prefix
+                )
+                upsert_to_collection(vnet, 'subnets', subnet_obj, 'name')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: no need  to check if the subnet exists?

Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: when the vnet_type is "new" (line 613), it make sure the subnet not exists, however the condition can't make sure vnet exists.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -674,7 +674,7 @@ def load_arguments(self, _):
         with self.argument_context(scope, arg_group='Network') as c:
             c.argument('vnet_name', help='Name of the virtual network when creating a new one or referencing an existing one.')
             c.argument('vnet_address_prefix', help='The IP address prefix to use when creating a new VNet in CIDR format.')
-            c.argument('subnet', help='The name of the subnet when creating a new VNet or referencing an existing one. Can also reference an existing subnet by ID. If omitted, an appropriate VNet and subnet will be selected automatically, or a new one will be created.')
+            c.argument('subnet', help='The name of the subnet when creating a new VNet or referencing an existing one. Can also reference an existing subnet by ID. If omitted, an appropriate VNet and subnet will be selected automatically, or a new one will be created when the specified VNet exits.')
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: typo `exits`

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -613,9 +613,30 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
...
+                vnet = cached_put(cmd, client.create_or_update, vnet, resource_group_name, vnet_name).result()
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: I am interested in cached_get, cached_put. What does cached mean

**Pull request closed without merging**